### PR TITLE
processor(ticdc): decouple processor from reactor

### LINKED ISSUE #9780: decouple processor from the reactor
decouple processor from the reactor

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #9780

### What is changed and how it works?
1. make the processor tick logic not depend on the reactor anymore.
2. add Processor interface

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/processor/manager.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ import (
 	"github.com/pingcap/tiflow/cdc/model"
 	"github.com/pingcap/tiflow/pkg/config"
 	cdcContext "github.com/pingcap/tiflow/pkg/context"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
```
Comment (CharlesCheung96) [inline cdc/processor/manager.go:28]:
```suggestion
  "github.com/pingcap/tiflow/pkg/errors"
```

Comment (CharlesCheung96) [reply cdc/processor/manager.go:28]: Ref: https://github.com/pingcap/tiflow/blob/master/pkg/errors/reexport.go

Comment (sdojjy) [reply cdc/processor/manager.go:28]: we imported "github.com/pingcap/errors" in this file too, must rename the package here

File: cdc/processor/processor.go [hunk 2]
```diff
@@ -444,15 +464,22 @@ func isProcessorIgnorableError(err error) bool {
...
-// the `state` parameter is sent by the etcd worker, the `state` must be a snapshot of KVs in etcd
+// the `info` parameter is sent by metadata store, the `info` must be the latest value snapshot.
+// the `status` parameter is sent by metadata store, the `status` must be the latest value snapshot.
 // The main logic of processor is in this function, including the calculation of many kinds of ts,
-// maintain table pipeline, error handling, etc.
+// maintain table components, error handling, etc.
 //
 // It can be called in etcd ticks, so it should never be blocked.
-func (p *processor) Tick(ctx cdcContext.Context) error {
+func (p *processor) Tick(ctx cdcContext.Context,
```
Comment (CharlesCheung96) [inline cdc/processor/processor.go:473]:
```suggestion
func (p *processor) Tick(
    ctx cdcContext.Context,
```

**Review by CharlesCheung96 — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**